jdaviz/tests/test_app.py

```diff
@@ -17,3 +17,58 @@ def test_get_tray_item_from_name():
 
     with pytest.raises(KeyError, match='not found in app'):
         app.get_tray_item_from_name('imviz-compass')
+
+
+def test_duplicate_data_labels(specviz_helper, spectrum1d):
+    specviz_helper.load_spectrum(spectrum1d, data_label="test")
+    specviz_helper.load_spectrum(spectrum1d, data_label="test")
+    dc = specviz_helper.app.data_collection
+    assert dc[0].label == "test"
+    assert dc[1].label == "test (1)"
+    specviz_helper.load_spectrum(spectrum1d, data_label="test_1")
+    specviz_helper.load_spectrum(spectrum1d, data_label="test")
+    assert dc[2].label == "test_1"
+    assert dc[3].label == "test (2)"
+
+
+def test_duplicate_data_labels_with_brackets(specviz_helper, spectrum1d):
+    specviz_helper.load_spectrum(spectrum1d, data_label="test[test]")
+    specviz_helper.load_spectrum(spectrum1d, data_label="test[test]")
+    dc = specviz_helper.app.data_collection
+    assert len(dc) == 2
+
+
+def test_return_data_label_is_none(specviz_helper):
+    data_label = specviz_helper.app.return_data_label(None)
+    assert data_label == "Unknown"
+
+
+def test_return_data_label_is_image(specviz_helper):
+    data_label = specviz_helper.app.return_data_label("data/path/test.jpg")
+    assert data_label == "test[jpg]"
+
+
+def test_hdulist_with_filename(cubeviz_helper, image_cube_hdu_obj):
+    image_cube_hdu_obj.file_name = "test"
+    data_label = cubeviz_helper.app.return_data_label(image_cube_hdu_obj)
+    assert data_label == "test[HDU object]"
+
+
+@pytest.mark.filterwarnings('ignore:.* is a low contrast image')
+def test_file_path_not_image(imviz_helper, tmp_path):
+    path = tmp_path / "myimage.fits"
+    path.mkdir()

```

Comment:
Do we need this?
```suggestion
```

Reply:
Yes.

Reply:
But why? pytest should have made the `tmp_path` directory for you.

Reply:
I'm not sure, but when I removed it the test failed.

Reply:
It did not parse the filename how I would've expected it to parse the filename based on what I have in `return_data_label`. It should return 'myimage` but when I did not include that line it kept the full file path.

Reply:
Ah, so you actually want "image.fits" to be a *sub-directory name* and that sub-directory must exist? What if someone simply passes in a path that does not end in `.fits` or that path does not really exist or the image does not exist?

I am not sure what you really want to test here. Can you please clarify?

Reply:
`path` needs to be a valid  file/directory that exists in order to test this conditional statement: https://github.com/spacetelescope/jdaviz/pull/1672/files#diff-33416aee3e2d4c0d86f96811f082119130c8598ef01ce1c9a7e852ecc39779b3R950

If you pass in a file that does not end in `.fits` but exists, it should go through that conditional as well. If the path does not exist, then it will move to the next conditional.

Reply:
Ok, I changed it. Ready for review now.